### PR TITLE
Enable processing from the registered schema in the ui package's renderer (for internal use only)

### PR DESCRIPTION
**Pull request by hand-dot**
ref: https://github.com/pdfme/pdfme/issues/246

### CODE REVIEW

**Comment by hand-dot** [comment]
@peteward @steffancarrington 
Could you check my changes?

File: packages/ui/src/components/SchemaUI.tsx
Comment (hand-dot) [inline packages/ui/src/components/SchemaUI.tsx:1]: I'll change filename to SchemaRenderer

File: packages/ui/src/class.ts [hunk 1]
```diff
@@ -79,22 +85,28 @@ export abstract class BaseUIClass {
...
     this.resizeObserver.observe(this.domContainer!);
 
+    const { lang, font } = options;
     if (lang) {
       this.lang = lang;
     }
     if (font) {
       this.font = font;
     }
+    // TODO: In the future, when we support custom schemas, we will create the registry using options.renderer instead of {}.
```
Comment (hand-dot) [inline packages/ui/src/class.ts:105]: set renderer

File: packages/ui/src/components/SchemaUI.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  const { schema, editable, onChange, stopEditing, tabIndex, placeholder } = props;
+
+  const ref = useRef<HTMLDivElement>(null);
+
+
+  useEffect(() => {
+    if (ref.current && schema.type) {
+      const schemaType = schema.type as string;
+      const renderer = rendererRegistry[schemaType];
```
Comment (hand-dot) [inline packages/ui/src/components/SchemaUI.tsx:49]: get renderer

File: packages/ui/src/components/SchemaUI.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      if (!renderer) {
+        console.error(`Renderer for type ${schema.type} not found`);
+        return;
+      }
+
+      ref.current.innerHTML = '';
+
+      renderer.render({
```
Comment (hand-dot) [inline packages/ui/src/components/SchemaUI.tsx:57]: use renderer

File: packages/generator/src/renders/text.ts [hunk 4]
```diff
@@ -31,8 +31,8 @@ import {
   convertSchemaDimensionsToPt
 } from '../renderUtils'
 
-const getFontProp = async ({ input, font, schema }: { input: string, font: Font, schema: TextSchema }) => {
-  const fontSize = schema.dynamicFontSize ? await calculateDynamicFontSize({ textSchema: schema, font, input }) : schema.fontSize ?? DEFAULT_FONT_SIZE;
+const getFontProp = async ({ value, font, schema }: { value: string, font: Font, schema: TextSchema }) => {
+  const fontSize = schema.dynamicFontSize ? await calculateDynamicFontSize({ textSchema: schema, font, input: value }) : schema.fontSize ?? DEFAULT_FONT_SIZE;
```
Comment (peteward) [inline packages/generator/src/renders/text.ts:35]: we could rename the arguments on `calculateDynamicFontSize` in future to align with this renaming 👍

File: packages/ui/src/renders/barcodes.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
+import type { RenderProps } from "../types"
```
Comment (peteward) [inline packages/ui/src/renders/barcodes.ts:1]: maybe the folder name for these files should be `renderers/` ?

**Review by peteward — APPROVED** [review]
looking great @hand-dot 
I am loving your work ❤️

**Pull request merged**